### PR TITLE
Build and deploy the book automatically on GHA

### LINKED ISSUE #315: Add GHA to build and deploy the book
Based on **bookdown** one

### PR DESCRIPTION
**Pull request by cderv**
closes #315

### CODE REVIEW

**Comment by cderv** [comment]
@yihui @emilyriederer here is a first try at auto building and deploying the book. 

What has been done: 

* Packages required are explicitly in DESCRIPTION
* GHA uses **pak** to install missing dependencies from DESCRIPTION and the R library is cached. 
* Some tweak inside the book (using python3 for example)
* At first I wanted to use **renv** but it does not handle the system dependencies the same and for GHA it seems easier to use **pak**. This allows us a to have a book that would be rendered using last version of packages. A nice way to test that everything is still working with last version of packages. 
* but **renv** works also because I set it to use explicit definition. 
* SETUP.md instruction has been updated
* I did not remove the Makefile part, but I thought about it. Not** sure it is so usual and it adds a requirement (mainly on Windows). Hopefully, new Rtools comes with a `make` that works for me so it is ok. 

The important part is that it is now automatically build. I am happy to hear feedback on the package management part 😅 

Another improvement would be to have PR preview for new content but this will be in another PR. (and it requires new env var to setup in the REPO for netlify).

Any thoughts before I merge ?

File: 18-references.Rmd [hunk 1]
```diff
@@ -90,12 +90,10 @@ pkgs <- c(
...
 )
 
-# install packages if missing
-lapply(pkgs, function(pkg) {
-    if (system.file(package = pkg) != '') return()
-    install.packages(pkg, repos = "https://cloud.r-project.org/") 
-    }
-)
+# Warn to add new deps in DESCRIPTION
+pkg2 <- setdiff(pkgs, c("stats", "graphics", "grDevices", "utils", "datasets", "methods", "base"))
```
Comment (yihui) [inline 18-references.Rmd:94]: I have an internal object `knitr:::.base.pkgs` that you may consider using here.

**Review by yihui — COMMENTED** [review]
Looks good! Thanks!

> * I did not remove the Makefile part, but I thought about it. Not sure it is so usual and it adds a requirement (mainly on Windows)

You can leave it behind. I don't feel it is an important issue.

> Another improvement would be to have PR preview for new content but this will be in another PR.

You can add it when anyone feels it is necessary. Personally I don't mind much---when I review book PRs, I often check out the PR locally and edit/preview it on my computer.

File: 18-references.Rmd (hunk 1 above)
Comment (cderv) [reply 18-references.Rmd:94]: Oh I look for such thing without success. Thanks ! 

```suggestion
pkg2 <- setdiff(pkgs, knitr:::.base.pkgs)
```

**Comment by cderv** [comment]
@emilyriederer I'll merge this so that the books build and deploy. If you have feedback, we can modify in a following up PR. Hope it is fine with you. Thanks

**Pull request merged**